docs/images/data_prefetch_architecture.drawio

```diff
@@ -0,0 +1 @@
+<mxfile host="app.diagrams.net" modified="2022-06-05T15:16:33.191Z" agent="5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.64 Safari/537.36" etag="9ug3HmWtyXvSUMf4NadX" version="18.0.6" type="device"><diagram id="WxfRxpbHqFga9GgLfe8u" name="Page-1">7Vzrc5s4EP9rPNP70AwChO2PeTTXublOO83cpf2UkY2MaTCiICfx/fUnCYmHMH6Vh33HlwStpEXa/WlZdhePrNvV2+8xipafiIuDkWm4byPrbmSawDYd9o9TNilloghe7LtyUE548P/BkmhI6tp3cVIaSAkJqB+ViXMShnhOSzQUx+S1PGxBgvJdI+ThCuFhjoIq9dF36VJSgTPNOz5i31tStb9x2rFCarDcSbJELnktkKwPI+s2JoSmV6u3Wxxw4Sm5pPPua3qzhcU4pIdMePQAer7/64v/7W9n/QXcP/iz4L3k8oKCtdzwTUBmXzFycTyyrkdcWxbThXVzNbo1R9dGFOMFpvPl04yNe3IRRU8xCpkMKyMTSqJ0gJoj5UA3SrgxWYcu5usDbB4KfC9k1wFesA3dvOCY+kwR15K88l2XT7x5XfoUP0Rozrm8Mtgx2oKEVIIH2Kwtt8VY4LdaeYFMCwy+mKwwjTdsiJxgQaliiVxomWn7tYCDqVTusgABR9KQhJ6Xsc61wy6kgo5QFqwIELsMrLJJYrokHglR8CGn3uQi5qrJx/xJSCTl/gNTupHCQ2tKGGlJV4HsxW8+/canX0HZ+i6Z8eu7t2Jjoxoh225hEm9+V/x4I58mWmredi3yTe7WIZMJWcdzvEN2UncUxR6m+w5EFRMxDhD1X8rraFzBZuU0BoSdgEWSHsWAH4tZzK48KqSTUpIIhSVYOD/XRAxF82dPAOD9nAQkPdCGH/rUR0HKQI5ULK+4xcUhjhHFXFoxP2S1N2Z7TO/d6npQFAWbc1lMzCwjWrI/3d32iC3XmlejxmwWzrl+9jRr3IBBNZ2yQXXsLQbV3GJQx20ZVKty3j4/PHR51kJGYdzWUcQsc5/Yrr/3xcPOmtpnBju7ArsYe37CWQ/Y+09hDypcnQ32nMGHPNmHtDryIeXUL8QPacGOWdrjE5plFun65azii6DGyB5rBtHQGKUbrDASeMv2czoExwMET4ag3S8EbQ2CzqkQnGgQBN1CcFJ5AvOwxy2aL7Ee9TCNPHxRIov4xta+cqBEhkcuKfQBzbKegQOrjy17y2Nr0tZja1pR2N16tdoUNKY5DJrist4tHkuEXNcPvdRRMaI3IUNjxRDoh2XiNjdKh0Frd3LQiqs7nCXR8KK52+OflOFrTkEVvlaXXpeKZhfwe+8H+P8LX906DmCutcXWAWDu9BUCVJMG98kA5QHK+6DsjM8OytWI+3WyCeePJH7+5MUanrdokwvtfSKkxtUI7BrUNAAL49B71foKzYRyYm/2zoTMIWQiN4oXvxWWuUArP0jDWUa2HMn2D0xvYuSHCev7REKi96cMV6wnScF6zOaXKGSucpqdzE8m/rnGCW1Yl5o4m3LRLkK+i6RWuhdmlKD2Tmub1RBdltLtJM0LrPMJkKie4wIkeUjksACJi5Jl9i7cYLRk2mu0xIYNBeygHi3pOGAHhsKD9jE46QSD0AHNYBAC0C0Gh8TF6RhU4Ye+QKgl/uH4VBBqqVyor6VtEA6pi18AIegXhNrLJ5ycCELH0EBodQzC7cmLz7Mfov51S81mITzxNCcr5rgnCQPkznHrcP/I+XIdPieVI3HWqQ1H1eceW9XZWmoDTM/HqFy2u6+iOX35+07ZMFinPuZ0RlBVjXRkYbKK9QGRnSGyrYeeBslTH3o6I6i/y7YNSXApkOwMWof69CkGG4eWoyVZ7emJ0Y3erZ15KdA6e2t3qIffFiSnmouvx8kuxtr1GgEGBUDm8NwDSVACZI7PviF56AO4nedvY0ZyYk36NZIyBFx46dyb3irntNA7I008af/qslq781N12fkRe6HTPkaFbJucKr60zFpq21BsnFHu+DX3fCHfGmQC3DcWZGMVNE9iY+ZsUulmPaqqghGuOYHLm12L/CUUG+atq3QFWd2DPk5qRnFNJca7VQ2EYi+EyjtEQ9ZCbO3MeVQXolVh9LsakfjkLZGSFbfRqjXy9aU43rHCSro2W41K0+9eUb0Kiz0CNGotxQHmlsVlsRZOMMTQQhyFE0UkhdOFoecUIJqZARYkW9DSUA0niGBN+cYy6nII+rOuDPi5RYXcpGUjHSs/KcYmp1s8YqzoaWQm5144XSpCk3WOC33MNGZ0r7AA/VyKZnY4i8SyyZDjKrZlZ3EFl/NW69hy5UgvVQT62qUB+yhqBuIOvyjK63ZqblYuT3NJpW6MzUtVN3zXeWy1uPooYF+xuA3r3alf+4zaGZyVwVkZnJXBWRmclZJtGXyVel+F51Dv8Iuf/8bEOfgpNW5I1Wf4Zd1tK9MfnKDDvzkCZScITKy+vaDxZXhBX9EiUZZ38IQGT2jwhAZPqFVPiBucr2fqCu16VgpDeWYewxA5adJpcHr3GS6mFm1nAvHURHmDiUflfe1NPKbBqvMtRrO0pDoYd5t5VN/pn38u/EBIjo5KlDcJSafXXLhtaMZufBogdT5g0jEgeylFa8XenWJ4mwQk7BWQUH9l13+E6ODPUvSfaACwIUSOpDNbGJ6/Ilsf/gU=</diagram></mxfile>
```

Comment:
Do we need to export drawio format to png, and reference it from somewhere (in doc or code)?

Reply:
We need more work to improve documentation. There are also coming work to document rafs v6 data format. We could handle it then.